### PR TITLE
Refactor popovers and color picker into anchored popover and menu

### PR DESCRIPTION
**Pull request by lsjroberts**
https://ag-grid.atlassian.net/browse/AG-12601

Refactors popovers into reusable components:

- Popover
  - AnchoredPopover
    - ColorPicker
    - Menu (annotation type picker, font size)
  - Dialog (future PR)
    - SettingsDialog (draggable popover for annotations text, etc)

### CODE REVIEW

**Comment by lsjroberts** [comment]
Don't merge just yet, it will likely cause conflicts for other devs. Will co-ordinate with them.

File: packages/ag-charts-enterprise/src/features/color-picker/colorPickerStyles.css [hunk 1]
```diff
@@ -24,6 +16,12 @@
     --checker: url('data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" width="4" height="4"><rect x="0" y="0" width="4" height="4" fill="%23fff"/><path d="M0 0H2V4H4V2H0Z" fill="%23b2b2b2"/></svg>');
 }
 
+.ag-charts-color-picker__content {
```
Comment (jacobp100) [inline packages/ag-charts-enterprise/src/features/color-picker/colorPickerStyles.css:19]: Did the layout change here?

File: packages/ag-charts-enterprise/src/features/color-picker/colorPickerStyles.css [hunk 2]
```diff
@@ -24,6 +16,12 @@
...
+    display: flex;
+    flex-direction: column;
+    color: var(--ag-charts-toolbar-foreground-color);
```
Comment (jacobp100) [inline packages/ag-charts-enterprise/src/features/color-picker/colorPickerStyles.css:22]: This is already in the popover styles, I believe

```suggestion
```

**Review by jacobp100 — APPROVED** [review]
Nice!

File: packages/ag-charts-enterprise/src/features/color-picker/colorPickerStyles.css (hunk 1 above)
Comment (lsjroberts) [reply packages/ag-charts-enterprise/src/features/color-picker/colorPickerStyles.css:19]: Yeah, it's been flattened a little, merging `.ag-charts-popover` and `.ag-charts-color-picker` onto the same element (removing the `colorPickerContainer` element which served the same purpose as the outer popover element).

Ideally the `__content` styles could be merged into `.ag-charts-color-picker` but I had some issues with the width of the sliders. Will double check that.

From:
Screenshot 2024-08-15 at 16 47 57

To:
Screenshot 2024-08-15 at 16 48 24

File: packages/ag-charts-enterprise/src/features/color-picker/colorPickerStyles.css (hunk 2 above)
Comment (lsjroberts) [reply packages/ag-charts-enterprise/src/features/color-picker/colorPickerStyles.css:22]: Woops, good spot, thanks!

File: packages/ag-charts-enterprise/src/components/popover/popover.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        this.moduleId = `popover-${id}`;
+
+        this.element = ctx.domManager.addChild(canvasOverlay, this.moduleId);
+        this.element.role = 'presentation';
+
+        this.destroyFns.push(() => ctx.domManager.removeChild(canvasOverlay, this.moduleId));
+    }
+
+    public show(options: Options) {
+        const popover = createElement('div');
```
Comment (jacobp100) [inline packages/ag-charts-enterprise/src/components/popover/popover.ts:39]: Maybe we could return this so things like the colour picker can attach css custom properties directly?

**Review by alantreadway — APPROVED** [review]
LGTM, merging as I know the intent is to continue iterating on this for the actual feature.

**Pull request merged**